simpledotorg/simple-server
# Progress period reports

**Story card:** [sc-7595](https://app.shortcut.com/simpledotorg/story/7595/add-progress-card-to-the-home-page-and-create-daily-and-monthly-progress-report-pages)

## Because

We will now show (behind a feature flag) the new "Registrations and follow-ups" card on the homepage. The "Daily" and "Monthly" reports will show the data we currently show in the Progress tab, but using the new UI designs.

## This addresses

**Original progress tab home page**
![image](https://user-images.githubusercontent.com/16785131/160690201-0d047c55-0308-4fee-8d42-b8602bec62ba.png)

**New progress tab home page**
![image](https://user-images.githubusercontent.com/16785131/160726405-dcf4541b-7b07-4126-877b-57e75811044e.png)

**New daily report**
![image](https://user-images.githubusercontent.com/16785131/160690284-a30756e0-68ac-46ff-ba3a-dfaab2b3705e.png)

**New monthly report**
![image](https://user-images.githubusercontent.com/16785131/160690329-cf7e0d28-139c-4e45-b48a-5a4fff9f202b.png)

## Test instructions

**1. Make sure the original home page works IF the `new_progress_tab` feature flag is not enabled:**
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Make sure the "Today" card shows data for past days.
* Make sure the "Registered patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Follow-up patients" card works. Click the dropdown and change the diagnosis and gender. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Make sure the "Hypertension controlled" card works. Tap the chart area and make sure the month labels show. Click the help icon and make sure the definition page, and only the definition page, shows up.
* Click the "Cohort reports" button and make sure the cohort report page shows up. When you click out of the page, make sure the home page appears.
* Make sure the "Achievements" card shows total follow-up and total registrations values that match the totals shown in the cards above. Also make sure the badges match the values. For example, if there are 27 follow-ups, there should be 1 completed badge and 4 empty badges.

**2. Make sure the new home page works IF the `new_progress_tab` feature flag is enabled:**
* Go to a facility progress report page (Reports > Select a facility report from the search bar > Progress report).
* Change your browser to a mobile view.
* Click the "Daily" report button. Make sure the "Last updated on" subtitle is showing. Click the dropdown and select a new date. Make sure the values in the table are updated.
* Go back to the home page and click the "Monthly" report button. Make sure the "Last updated on" subtitle is showing. Click the registrations dropdown and make sure the table is updated as you select new values. Do the same for the follow-ups dropdown.
* Go back to the home page and click the "Hypertension" report button. Scroll down the page and click on the patients initiated on treatment charts and help icons — make sure the charts work properly and the tooltips show the right data.
* Continue scrolling down and make sure the cohort report charts work properly and the tooltips show the right data.
* Go back to the home page and scroll down to the "Achievements" card. The registrations and follow-ups values should match the totals shown in the "Monthly" report.
